sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -512,6 +512,9 @@ class Analyzer(
 
       // A special case for Generate, because the output of Generate should not be resolved by
       // ResolveReferences. Attributes in the output will be resolved by ResolveGenerate.
+      case g @ Generate(generator, _, _, _, _, _)
+        if !g.resolved && generator.resolved => g
+
       case g @ Generate(generator, join, outer, qualifier, output, child)
```

Comment:
`if child.resolved` is guaranteed at the beginning of this rule:
`case p: LogicalPlan if !p.childrenResolved => p`